.github/workflows/tests.yml

```diff
@@ -0,0 +1,55 @@
+name: Tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches: ["master"]
+    paths:
+      - "src/**"
+      - "Cargo.*"
+      - build.rs
+  pull_request:
+    branches: ["master"]
+    paths:
+      - ".github/workflows/unit-tests.yml"
+      - "src/**"
+      - "Cargo.*"
+      - build.rs
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  unit-tests:
+    runs-on: ${{ matrix.os }}
+
+    continue-on-error: ${{ matrix.rust == 'nightly' }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        rust: [1.70.0, stable, beta, nightly]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - run: rustup toolchain install ${{ matrix.rust }} --profile minimal
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Install cargo-hack
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: cargo install cargo-hack
```

Comment:
Are we planning to use `cargo-hack` for any of the checks ?

Reply:
No, it's just a leftover. I'll finalize it in further comments.